src/facade/parsed_command.h

```diff
@@ -0,0 +1,59 @@
+// Copyright 2025, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#pragma once
+
+#include "common/backed_args.h"
+#include "facade/memcache_parser.h"
+
+namespace facade {
+
+class ConnectionContext;
+class SinkReplyBuilder;
+
+// ParsedCommand is a protocol-agnostic holder for parsed request state.
+// It wraps cmn::BackedArguments so the facade can populate RESP arguments and
+// optionally attach a MemcacheParser::Command, complementing the arguments
+// with memcache-specific data.
+// The purpose of ParsedCommand is to hold the entire state of a parsed request
+// during its lifetime, from parsing to dispatching and reply building including
+// any async dispatching.
+class ParsedCommand : public cmn::BackedArguments {
+ protected:
+  SinkReplyBuilder* rb_ = nullptr;  // either RedisReplyBuilder or MCReplyBuilder
+  ConnectionContext* conn_cntx_ = nullptr;
+
+  std::unique_ptr<MemcacheParser::Command> mc_cmd_;  // only for memcache protocol
+
+ public:
+  ParsedCommand() = default;
+  ParsedCommand(SinkReplyBuilder* rb, ConnectionContext* conn_cntx)
+      : rb_{rb}, conn_cntx_{conn_cntx} {
+  }
+
+  void CreateMemcacheCommand() {
+    mc_cmd_ = std::make_unique<MemcacheParser::Command>();
+    mc_cmd_->backed_args = this;
+  }
+
+  void Init(SinkReplyBuilder* rb, ConnectionContext* conn_cntx) {
+    rb_ = rb;
+    conn_cntx_ = conn_cntx;
+  }
```

Comment:
why do you use it instead of constructor? To have a pool with recycling? Then why the constructor?